core/src/test/scala/org/apache/spark/HeartbeatReceiverSuite.scala

```diff
@@ -73,7 +73,7 @@ class HeartbeatReceiverSuite
       .setMaster("local[2]")
       .setAppName("test")
       .set(DYN_ALLOCATION_TESTING, true)
-    sc = spy(new SparkContext(conf))
+    sc = spy[SparkContext](new SparkContext(conf))
```

Comment:
Why change it? The following error will be reported if not changed:
<img width="1243" alt="image" src="https://github.com/apache/spark/assets/15246973/280bf2c4-1077-41cd-97d8-07e03df7ac12">